### ISSUE TITLE
AssertJ resource stubs are missing type parameters

### ISSUE THREAD

**Issue by tkvangorder** [openrewrite/rewrite-testing-frameworks#236]
Originally when the JUnit -> AssertJ recipes were created, the type information available on rewrite's ASTs was limited. However, now that the type system is complete, it is important to generate types that reflect the generic type parameters (especially with AssertJ).

The lack of type parameters in the class/method stubs results in the wrong types being used after the assertJ recipes have been run. This is rooted in how we generate the stubs using the [rewite-template-generator](https://github.com/openrewrite/rewrite-template-generator). This utility is using ASM to generate the stubs and most of the needed information is present to generate the classes and methods with type parameters.

The AssertJ recipes correctly transform the source code but the type information of the underlying method invocations will be non-null but will also be incorrect.

[Please see this class, as it is a good reference when dealing with ASM's descriptors.](https://github.com/openrewrite/rewrite/blob/5553659f2540b5db2bdb4963c1140a6dc8dac5a7/rewrite-java/src/main/java/org/openrewrite/java/typegeneration/AsmUtils.java)

**Comment by timtebeek** [comment]
I don't think this still applies, as much has changed since and we've been seeing good results with chained AssertJ migrations.
